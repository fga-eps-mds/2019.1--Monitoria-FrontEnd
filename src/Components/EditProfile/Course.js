import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
  },
  selectEmpty: {
    marginTop: theme.spacing.unit * 2,
  },
});

class SimpleSelect extends React.Component {
  state = {
    selectedCourse:0,
    labelWidth: 0,
  };

  componentDidMount() {
    this.setState({
      labelWidth: ReactDOM.findDOMNode(this.InputLabelRef).offsetWidth,
    });
  }

  handleChange = event => {
    this.setState({ selectedCourse: event.target.value });
  };

  render() {
    const { classes } = this.props;

    return (
      <form className={classes.root} autoComplete="off">
        
        <FormControl variant="outlined" className={classes.formControl}>
          <InputLabel
            ref={ref => {
              this.InputLabelRef = ref;
            }}
            htmlFor="outlined-age-simple"
          >
            Curso
          </InputLabel>
          <Select
            value={this.state.selectedCourse}
            onChange={this.handleChange}
            input={
              <OutlinedInput
                labelWidth={this.state.labelWidth}
                id="course"
              />
            }
          >
            <MenuItem value={0}>Engenharias</MenuItem>
            <MenuItem value={1}>Engenharia de Software</MenuItem>
            <MenuItem value={2}>Engenharia Aeroespacial</MenuItem>
            <MenuItem value={3}>Engenharia Eletr√¥nica</MenuItem>
            <MenuItem value={4}>Engenharia Automotiva</MenuItem>
            <MenuItem value={5}>Engenharia de Energia</MenuItem>        
          </Select>
        </FormControl>
      </form>
    );
  }
}

SimpleSelect.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(SimpleSelect);