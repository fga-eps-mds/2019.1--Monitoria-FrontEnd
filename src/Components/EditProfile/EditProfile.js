import React, { Component } from 'react';
import { Grid, Button, TextField } from '@material-ui/core' ;
import BottomBar from '../Tab/Tab.js';
import AppBar from '../AppBar/AppBar.js';
import Pp from '../../Assets/img/Pp.png';
import Course from './Course.js';
import { Link } from 'react-router-dom';


class EditProfile extends Component {
    state = {
        name:'batatinha',
        course: 1,
        telgram: '',
    }

    editProfile = () =>{
        const {name,course,telegram} = this.state
        let newUserData  = {name: name, course: course, telegram: telegram}

    }    
  render() {
    return (
        <div style={{overflowX:'hidden'}}>
            <div> 
                <Grid container justify="center" alignItems="stretch">
                    <AppBar/>
                </Grid> 
            </div>
        <div>
            <Grid container justify="center" alignContent="center" alignItems="center">
                <Grid item> 
                    <img src={Pp} alt="Profilepic" style={{width: 130,height:130,padding:80}} ></img>
                </Grid>
            </Grid>
            <Grid container justify="center" alignContent="center" alignItems="center" direction="column" >
                <Grid item xs={12}> 
                    <TextField
                        id="name"
                        label="Nome"
                        multiline
                        Maxrows="4"
                        margin="normal"
                        defaultValue={this.state.name}
                        onChange={(event)=>this.setState({
                            name: event.target.value,
                        })}
                    />
                </Grid>
                <Grid item> 
                    <TextField
                        id="telegram"
                        label="Telegram"
                        multiline
                        Maxrows="4"
                        placeholder="@"
                        margin="normal"
                        onChange={(event)=>this.setState({
                            telegram: event.target.value,
                        })}
                    />
                </Grid>
                <Grid item style={{padding:30}}>
                    <Course/>
                </Grid>
            </Grid>
            <Grid container justify="center" alignContent="center" alignItems="center" direction="row" spacing={24}>
                <Grid item>
                    <Button component={Link} to="/EditProfile" variant="outlined" onClick={this.editProfile} color="primary">
                        Confirmar
                    </Button>
                </Grid>
                <Grid item>
                    <Button component={Link} to="/Feed" variant="outlined" color="primary">
                        Cancelar
                    </Button>
                </Grid>
            </Grid>
            <Grid> 
                <Grid >
                <BottomBar>
                </BottomBar>
                </Grid>
            </Grid>
            </div>
        </div>
        
    );   
  }
}

export default EditProfile;
