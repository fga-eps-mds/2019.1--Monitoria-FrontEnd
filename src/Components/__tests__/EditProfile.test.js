import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import EditProfile from '../EditProfile/EditProfile';

const profile = jest.fn();

test('Logout renders a snapshot properly', () => {
  const tree = renderer.create(
    <Router><EditProfile profile={profile}/></Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});