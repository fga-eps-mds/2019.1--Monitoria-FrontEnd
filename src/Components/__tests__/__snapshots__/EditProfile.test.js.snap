// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders a snapshot properly 1`] = `
<div>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  >
    <div
      className="SimpleAppBar-root-98"
    >
      <div
        className="MuiGrid-container-1"
      >
        <header
          className="MuiPaper-root-108 MuiPaper-elevation4-114 MuiAppBar-root-99 MuiAppBar-positionStatic-103 MuiAppBar-colorPrimary-106"
          style={
            Object {
              "background": "#1DA1F2",
            }
          }
        >
          <div
            className="MuiToolbar-root-135 MuiToolbar-regular-137 MuiToolbar-gutters-136"
          >
            <img
              alt="Logo"
              height="40"
              src="Logo.png"
              width="40"
            />
            <h6
              className="MuiTypography-root-139 MuiTypography-h6-156 MuiTypography-colorInherit-168"
            >
              +Monitoria
            </h6>
          </div>
        </header>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-direction-xs-column-4 MuiGrid-align-items-xs-center-9 MuiGrid-align-content-xs-center-13 MuiGrid-justify-xs-center-18"
    style={
      Object {
        "padding": 100,
      }
    }
  >
    <div
      auto-md={true}
      className="MuiGrid-item-2"
    >
      <img
        alt="pp"
        height="120"
        src="Pp.png"
        width="120"
      />
    </div>
    <div
      auto-md={true}
      className="MuiGrid-item-2"
    >
      <button
        className="MuiButtonBase-root-201 MuiButton-root-175 MuiButton-outlined-183 MuiButton-outlinedPrimary-184 MuiButton-sizeSmall-198"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-176"
        >
          Alterar foto
        </span>
        <span
          className="MuiTouchRipple-root-319"
        />
      </button>
    </div>
    <div
      className="MuiGrid-item-2"
      md-auto={true}
    >
      <div
        className="MuiFormControl-root-204 MuiFormControl-marginNormal-205"
      >
        <label
          className="MuiFormLabel-root-219 MuiInputLabel-root-208 MuiInputLabel-formControl-213 MuiInputLabel-animated-216"
          data-shrink={false}
          htmlFor="nomeTextField"
        >
          Nome
        </label>
        <div
          className="MuiInputBase-root-239 MuiInput-root-226 MuiInput-underline-230 MuiInputBase-formControl-240 MuiInput-formControl-227"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-249 MuiInput-input-234"
            disabled={false}
            id="nomeTextField"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2"
      md-auto={true}
    >
      <div
        className="MuiFormControl-root-204 MuiFormControl-marginNormal-205"
      >
        <label
          className="MuiFormLabel-root-219 MuiInputLabel-root-208 MuiInputLabel-formControl-213 MuiInputLabel-animated-216"
          data-shrink={false}
          htmlFor="emailTextField"
        >
          Email
        </label>
        <div
          className="MuiInputBase-root-239 MuiInput-root-226 MuiInput-underline-230 MuiInputBase-formControl-240 MuiInput-formControl-227"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="email"
            className="MuiInputBase-input-249 MuiInput-input-234 MuiInputBase-inputType-252 MuiInput-inputType-237"
            disabled={false}
            id="emailTextField"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="email"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2"
      md-auto={true}
    >
      <form
        autoComplete="off"
        className="SimpleSelect-root-256"
      >
        <div
          className="MuiFormControl-root-204 SimpleSelect-formControl-257"
        >
          <label
            className="MuiFormLabel-root-219 MuiInputLabel-root-208 MuiInputLabel-formControl-213 MuiInputLabel-animated-216 MuiInputLabel-outlined-218"
            data-shrink={false}
            htmlFor="outlined-age-simple"
          >
            Curso
          </label>
          <div
            className="MuiInputBase-root-239 MuiOutlinedInput-root-266 MuiInputBase-formControl-240"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-279 MuiOutlinedInput-notchedOutline-273"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-280"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-259"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-260 MuiSelect-selectMenu-263 MuiSelect-outlined-262 MuiInputBase-input-249 MuiOutlinedInput-input-274"
                id="select-age"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                id="outlined-age-simple"
                name="age"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-281 MuiSelect-icon-265"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="MuiGrid-item-2"
      md-auto={true}
    >
      <div
        className="MuiFormControl-root-204 MuiFormControl-marginNormal-205"
      >
        <label
          className="MuiFormLabel-root-219 MuiInputLabel-root-208 MuiInputLabel-formControl-213 MuiInputLabel-animated-216"
          data-shrink={false}
          htmlFor="telefone    TextField"
        >
          Telefone
        </label>
        <div
          className="MuiInputBase-root-239 MuiInput-root-226 MuiInput-underline-230 MuiInputBase-formControl-240 MuiInput-formControl-227"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-249 MuiInput-input-234"
            disabled={false}
            id="telefone    TextField"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2"
    >
      <a
        className="MuiButtonBase-root-201 MuiButton-root-175 MuiButton-outlined-183 MuiButton-outlinedPrimary-184"
        href="/Feed"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-176"
        >
          Alterar
        </span>
        <span
          className="MuiTouchRipple-root-319"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className=""
    >
      <div
        className="MuiPaper-root-108 MuiPaper-elevation2-112 IconLabelTabs-root-294"
      >
        <div
          className="MuiTabs-root-295"
        >
          <div
            className="MuiTabs-flexContainer-296"
          >
            <div
              className="MuiTabs-scroller-298 MuiTabs-fixed-299"
              onScroll={[Function]}
              role="tablist"
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              <div
                className="MuiTabs-flexContainer-296"
              >
                <a
                  aria-selected={true}
                  className="MuiButtonBase-root-201 MuiTab-root-304 MuiTab-textColorSecondary-308 MuiTab-selected-309 MuiTab-fullWidth-311"
                  href="/Feed"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-312"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-281"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <svg>
                        <path
                          d="M10,20V14H14V20H19V12H22L12,3L2,12H5V20H10Z"
                          fill="#1da1f2"
                        />
                      </svg>
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-319"
                  />
                </a>
                <a
                  aria-selected={false}
                  className="MuiButtonBase-root-201 MuiTab-root-304 MuiTab-textColorSecondary-308 MuiTab-fullWidth-311"
                  href="/EditProfile"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-312"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-281"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <svg>
                        <path
                          d="M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,14C16.42,14 20,15.79 20,18V20H4V18C4,15.79 7.58,14 12,14Z"
                          fill="#1da1f2"
                        />
                      </svg>
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-319"
                  />
                </a>
                <a
                  aria-selected={false}
                  className="MuiButtonBase-root-201 MuiTab-root-304 MuiTab-textColorSecondary-308 MuiTab-fullWidth-311"
                  href="/Ranking"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-312"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-281"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,2H4V4L9.81,8.36C6.14,9.57 4.14,13.53 5.35,17.2C6.56,20.87 10.5,22.87 14.19,21.66C17.86,20.45 19.86,16.5 18.65,12.82C17.95,10.71 16.3,9.05 14.19,8.36L20,4V2M14.94,19.5L12,17.78L9.06,19.5L9.84,16.17L7.25,13.93L10.66,13.64L12,10.5L13.34,13.64L16.75,13.93L14.16,16.17L14.94,19.5Z"
                        fill="#1da1f2"
                      />
                       
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-319"
                  />
                </a>
                <a
                  aria-selected={false}
                  className="MuiButtonBase-root-201 MuiTab-root-304 MuiTab-textColorSecondary-308 MuiTab-fullWidth-311"
                  href="/Search"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-312"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-281"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <svg>
                        <path
                          d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
                          fill="#1da1f2"
                        />
                      </svg>
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-319"
                  />
                </a>
              </div>
              <span
                className="MuiPrivateTabIndicator-root-316 MuiPrivateTabIndicator-colorPrimary-317 MuiTabs-indicator-303"
                style={
                  Object {
                    "left": 0,
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
